### PR TITLE
allowlist for source containers using lookaside cache

### PR DESCRIPTION
**Pull request by rcerven**
* CLOUDBLD-9969

Signed-off-by: Robert Cerven <rcerven@redhat.com>

# Maintainers will complete the following section

- [x] Commit messages are descriptive enough
- [x] Code coverage from testing does not decrease and new code is covered
- [x] Python type annotations added to new code
- n/a JSON/YAML configuration changes are updated in the relevant schema
- n/a Changes to metadata also update the documentation for the metadata
- n/a Pull request has a link to an osbs-docs PR for user documentation updates
- [x] New feature can be disabled from a configuration file

### CODE REVIEW

**Review by hjmodi — APPROVED** [review]

File: atomic_reactor/schemas/config.json [hunk 1]
```diff
@@ -695,6 +695,10 @@
               "description": "Url with denylist yaml file, which will be used to exclude sources from cachito ",
               "type": "string"
           },
+          "allowlist_cache": {
+              "description": "Url with allowlist yaml file, which allows usage of lookaside cache",
```
Comment (chmeliik) [inline atomic_reactor/schemas/config.json:699]: I would prefer something like "lookaside_cache_allowlist", the current name sounds like something that would specify if/where/how to cache some allowlist

File: atomic_reactor/plugins/pre_fetch_sources.py [hunk 2]
```diff
@@ -194,12 +211,44 @@ def check_lookaside_cache_usage(self):
...
+            return
+
+        allowlist_cache_yaml = self._get_cache_allowlist()
+        should_raise = True
+
+        if allowlist_cache_yaml:
+            build_component = self.koji_build['package_name']
+            build_task = self.koji_build['extra'].get('container_koji_task_id')
+            task_info = self.session.getTaskInfo(build_task, request=True)
+            build_target = task_info['request'][1]
```
Comment (chmeliik) [inline atomic_reactor/plugins/pre_fetch_sources.py:231]: Is it expected there might not be a 'container_koji_task_id' in `koji_build['extra']`? Is calling `getTaskInfo(None, ...)` what we want to do in that case?

If it's not expected, could we do `koji_build['extra']['container_koji_task_id']` instead?

File: atomic_reactor/plugins/pre_fetch_sources.py [hunk 3]
```diff
@@ -186,6 +187,22 @@ def set_koji_image_build_data(self):
         if not self.koji_build_nvr:
             self.koji_build_nvr = self.koji_build['nvr']
 
+    def _get_cache_allowlist(self):
```
Comment (chmeliik) [inline atomic_reactor/plugins/pre_fetch_sources.py:190]:
```suggestion
    def _get_cache_allowlist(self) -> List[Dict[str, Any]]:
```

File: atomic_reactor/schemas/config.json (hunk 1 above)
Comment (rcerven) [reply atomic_reactor/schemas/config.json:699]: ok

File: atomic_reactor/plugins/pre_fetch_sources.py (hunk 2 above)
Comment (rcerven) [reply atomic_reactor/plugins/pre_fetch_sources.py:231]: right it has to be there

File: atomic_reactor/plugins/pre_fetch_sources.py (hunk 3 above)
Comment (rcerven) [reply atomic_reactor/plugins/pre_fetch_sources.py:190]: sure

**Review by chmeliik — APPROVED** [review]
LGTM :+1:

Might be good to document the new config option later

**Comment by rcerven** [comment]
> LGTM +1
> 
> Might be good to document the new config option later

it is in the schema, and I don't think cache is used upstream at all

**Comment by chmeliik** [comment]
> > LGTM +1
> > Might be good to document the new config option later
> 
> it is in the schema, and I don't think cache is used upstream at all

Fair point, waiving docs then

**Comment by rcerven** [comment]
rn updated: allowlist for source containers using lookaside cache

**Pull request merged**